src/ui/music.js

```diff
@@ -0,0 +1,175 @@
+                prereq: "NONE",
+                unlocked: true,
+            },
+            {
+                name: "scape-main",
+                prereq: "NONE",
+                unlocked: true,
+            },
+            {
+                name: "still-night",
+                prereq: CONSTANTS.PREREQUISITES.VARROCK_MINE,
+                unlocked: false,
+            },
+            {
+                name: "the-trade-parade",
+                prereq: "NONE",
+                unlocked: true,
+            },
+        ];
+
+        // Add song names to scroll window
+        this.songTexts = songNames.map((song) => {
+            const font = song.unlocked ? FONTS.SONG_UNLOCKED : FONTS.SONG_LOCKED;
+            let textObj = this.scrollWindow.add
+                .text(0, 0, "", font)
+                .setInteractive()
+                .on("pointerup", () => {
+                    if (textObj.unlocked) {
+                        this.audio.playBgm(song.name);
+                    }
+                });
+
+            // Store necessary data within the object itself
+            textObj.text = prettyPrintDash(song.name);
+            textObj.unlocked = song.unlocked;
+            textObj.prereq = song.prereq;
+
+            return textObj;
+        });
+        this.scrollWindow.addObjects(this.songTexts);
+        this.updateSongUnlocks();
+
+        // Add num unlocks at the bottom
+        this.numUnlockedText = dashboard.add
+            .text(633, 452, "0", FONTS.SONG_COUNT)
+            .setDepth(2)
+            .setOrigin(1, 0);
+        this.totalText = dashboard.add
+            .text(645, 452, "0", FONTS.SONG_COUNT)
+            .setDepth(2)
+            .setOrigin(0, 0);
+
+        // Add current playing song
+        this.currentSongText = dashboard.add
+            .text(555, 239, "NONE", FONTS.SONG_UNLOCKED)
+            .setDepth(2);
+        this.updateCurrentSongText();
+
+        // Add button / panel
+        this.panel = dashboard.add
+            .image(546, 205, "music-panel")
+            .setOrigin(0, 0)
+            .setDepth(1);
+        this.button = dashboard.add
+            .image(725, 466, "music-button")
+            .setOrigin(0, 0)
+            .setDepth(2)
+            .setInteractive()
+            .on("pointerdown", () => {
+                this.show(true);
+            });
+
+        // Setup destructor
+        dashboard.events.once("shutdown", () => {
+            dashboard.scene.remove(this.scrollWindow.name);
+        });
+    }
+
+    update() {
+        this.updateCurrentSongText();
+        this.updateSongUnlocks();
+    }
+
+    updateSongUnlocks() {
+        // Update song name color
+        this.songTexts.forEach((textObj) => {
+            if (!textObj.unlocked && characterData.getQuestCompleted(textObj.prereq)) {
+                textObj.unlocked = true;
+                textObj.setStyle(FONTS.SONG_UNLOCKED);
+                console.log("unlocked", textObj.text);
```

Comment:
Thoughts on logging something to chat window on song unlock?

Reply:
Sounds good to me

Reply:
Done https://github.com/EthanVieira/rs-clicker/pull/53/commits/3f36a73bc262e3b3a2cb541cbbbdfc4e65822ed3